codalib/xsdatetime.py

```diff
@@ -0,0 +1,189 @@
+from datetime import datetime, tzinfo, timedelta
+
+from pytz import timezone, utc as utc_tz
+
+# Constants for time parsing/formatting
+# This is a stub
+XSDT_FMT = "%Y-%m-%dT%H:%M:%S"
+# This should never change
+XSDT_TZ_OFFSET = 19
+# The str value to use for default local timezone
+# In localization operations
+DEFAULT_LOCAL_TZ = "US/Central"
+
+
+class InvalidXSDateTime(Exception):
+    pass
+
+
+class XSDateTimezone(tzinfo):
+    """
+    Concrete subclass of tzinfo for making sense of timezone offsets.
+    Not really worried about localization here, just +/-HHMM
+    """
+
+    def __init__(self, hours=0, minutes=0, sign=1):
+        self.minutes = hours * 60 + minutes
+        self.minutes *= sign
+
+    def utcoffset(self, dt):
+        return timedelta(minutes=self.minutes)
+
+    def dst(self, dt):
+        return timedelta(0)
+
+
+def xsDateTime_parse(xdt_str, local_tz=None):
+    """
```

Comment:
In the three functions where you have `local_tz` as a parameter, would it make sense to set the default value to `DEFAULT_LOCAL_TZ`, since that is what you end up doing when the value comes in as `None`?